allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,190 @@
+
+                else:
+                    field = LabelField(
+                        inputs[0][feature], label_namespace=feature, skip_indexing=True
+                    )
+
+            elif isinstance(value, Sequence):
+                # datasets Sequence of strings to ListField of TextField
+                if value.feature.dtype == "string":
+                    field_list = list()
+                    for item in inputs[0][feature]:
+                        item_field = TextField([Token(item)])
+                        field_list.append(item_field)
+
+                    if len(field_list) == 0:
+                        continue
+                    field = ListField(field_list)
+
+                # datasets Sequence of strings to ListField of LabelField
+                elif isinstance(value.feature, ClassLabel):
+                    field_list = list()
+                    for item in inputs[0][feature]:
+                        item_field = LabelField(
+                            label=item, label_namespace=feature, skip_indexing=True
+                        )
+                        field_list.append(item_field)
+                    if len(field_list) == 0:
+                        continue
+                    field = ListField(field_list)
+
+            # datasets.Translation cannot be mapped directly
+            # but it's dict structure can be mapped to a ListField of 2 ListField
+            elif isinstance(value, Translation):
+                if value.dtype == "dict":
+                    input_dict = inputs[0][feature]
+                    langs = list(input_dict.keys())
+                    field_langs = [LabelField(lang, label_namespace="languages") for lang in langs]
+                    langs_field = ListField(field_langs)
+                    texts = list()
+                    for lang in langs:
+                        texts.append(TextField([Token(input_dict[lang])]))
+                    field = ListField([langs_field, ListField(texts)])
+
+            # datasets.TranslationVariableLanguages
+            # is functionally a pair of Lists and hence mapped to a ListField of 2 ListField
+            elif isinstance(value, TranslationVariableLanguages):
+                if value.dtype == "dict":
+                    input_dict = inputs[0][feature]
+                    langs = input_dict["language"]
+                    field_langs = [LabelField(lang, label_namespace="languages") for lang in langs]
+                    langs_field = ListField(field_langs)
+                    texts = list()
+                    for lang in langs:
+                        index = langs.index(lang)
+                        texts.append(TextField([Token(input_dict["translation"][index])]))
+                    field = ListField([langs_field, ListField(texts)])
+
+            else:
+                raise ValueError(f"Datasets feature type {type(value)} is not supported yet.")
+
+        if field:
```

Comment:
```suggestion
        if field is not None:
```
There are many things in Python that are falsy. This makes clear which one you mean.